src/java/com/google/cloud/bigquery/dwhassessment/hooks/logger/EventLogger.java

```diff
@@ -0,0 +1,241 @@
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.UUID;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryPlan;
+import org.apache.hadoop.hive.ql.hooks.HookContext;
+import org.apache.hadoop.hive.ql.hooks.HookContext.HookType;
+import org.apache.hive.common.util.ShutdownHookManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Main handler of query events. Processes the captured events and writes resulting records to a
+ * file.
+ */
+public class EventLogger {
+  private static final Logger LOG = LoggerFactory.getLogger(EventLogger.class);
+
+  private static final int MAX_RETRIES = 2;
+  private static final int SHUTDOWN_WAIT_TIME = 5;
+  private static final int QUERY_EVENTS_QEUUE_DEFAULT_SIZE = 64;
+  private static final Duration DEFAULT_ROLLOVER_TIME_MILLISECONDS = Duration.of(600, SECONDS);
+
+  private final DatePartitionedLogger logger;
+  private final EventRecordConstructor eventRecordConstructor;
+  private final ScheduledThreadPoolExecutor logWriter;
+  private final int queueCapacity;
+  private final UUID loggerId;
+
+  private int logFileCount = 0;
+  private RecordsWriter writer;
+  private LocalDate writerDate;
+
+  // Singleton using DCL.
+  private static volatile EventLogger instance;
+
+  public static EventLogger getInstance(HiveConf conf, Clock clock) {
+    if (instance == null) {
+      synchronized (EventLogger.class) {
+        if (instance == null) {
+          instance = new EventLogger(conf, clock);
+          ShutdownHookManager.addShutdownHook(instance::shutdown);
+        }
+      }
+    }
+    return instance;
+  }
+
+  protected EventLogger(HiveConf conf, Clock clock) {
+    eventRecordConstructor = new EventRecordConstructor(clock);
+    this.loggerId = UUID.randomUUID();
+    this.queueCapacity =
+        conf.getInt(
+            HIVE_QUERY_EVENTS_QUEUE_CAPACITY.getConfName(), QUERY_EVENTS_QEUUE_DEFAULT_SIZE);
+    String baseDir = conf.get(HIVE_QUERY_EVENTS_BASE_PATH.getConfName());
+
+    if (StringUtils.isBlank(baseDir)) {
+      baseDir = null;
+      LOG.error(
+          "Log dir configuration key '{}' is not set, logging disabled.",
+          HIVE_QUERY_EVENTS_QUEUE_CAPACITY.getConfName());
+    }
+
+    this.logger = createLogger(baseDir, conf, clock);
+    if (logger == null) {
+      logWriter = null;
+      return;
+    }
+
+    ThreadFactory threadFactory =
+        new ThreadFactoryBuilder()
+            .setDaemon(true)
+            .setNameFormat("Migration Assessment Hook Query Log Writer %d")
+            .build();
+    logWriter = new ScheduledThreadPoolExecutor(1, threadFactory);
+
+    long rolloverInterval =
```

Comment:
I would suggest to add a time unit to the variable name.

Reply:
Done